test/internet_connection_test.dart

```diff
@@ -21,6 +23,55 @@ void main() {
       expect(await checker.hasInternetAccess, false);
     });
 
+    test('hasInternetAccess invoke responseStatusFn to determine success',
+        () async {
+      await TestHttpClient.run((client) async {
+        client.responseBuilder =
+            (req) => TestHttpClient.createResponse(statusCode: 500);
```

Comment:
Why is this code necessary? The tests are running properly without these additional Mock codes.

Reply:
This code to make sure we didn't connect to internet. Assuming we have air gapped CI env.